### PR TITLE
Add %site_title% and %site_description% macros

### PR DESCRIPTION
**Pull request by arnaudbroes**
#1271.

### CODE REVIEW

**Comment by michaeltorbert** [comment]
This should have unit testing for the old macros as well as the new in order to verify the behavior.

**Comment by arnaudbroes** [comment]
We should have for each and another to compare them and see if the result is the same.

**Comment by wpsmort** [comment]
This is tested and is good.

**Comment by arnaudbroes** [comment]
@EkoJR reach out to me first before looking at this since I'll need to add a unit test.

File: aioseop_class.php [hunk 1]
```diff
@@ -919,6 +919,9 @@ public function get_title_format( $args ) {
 		$w            = $info['w'];
 		$p            = $info['p'];
 
+		if ( strpos( $title_format, '%site_title%' ) !== false ) {
```
Comment (EkoJR) [inline aioseop_class.php]: The parent class method for strpos should be used instead. (multiple occurances)

File: aioseop_class.php [hunk 2]
```diff
@@ -919,6 +919,9 @@ public function get_title_format( $args ) {
...
+			$title_format = str_replace( '%site_title%', get_bloginfo( 'name' ), $title_format );
+		}
 		if ( strpos( $title_format, '%blog_title%' ) !== false ) {
```
Comment (EkoJR) [inline aioseop_class.php]: It might help to add a dev note here that %blog_title% is deprecated.

File: aioseop_class.php [hunk 3]
```diff
@@ -1462,7 +1465,13 @@ function get_aioseop_title( $post, $use_original_title_format = true ) {
 		} elseif ( is_search() && isset( $s ) && ! empty( $s ) ) {
 			$search = esc_attr( stripslashes( $s ) );
 			$title_format = $aioseop_options['aiosp_search_title_format'];
-			$title        = str_replace( '%blog_title%', $this->internationalize( get_bloginfo( 'name' ) ), $title_format );
+			$title        = str_replace( '%site_title%', $this->internationalize( get_bloginfo( 'name' ) ), $title_format );
+			if ( strpos( $title, '%blog_title%' ) !== false ) {
```
Comment (EkoJR) [inline aioseop_class.php]: The parent class method for strpos should be used instead. (multiple occurances)

Using Yoda operations is also a good habit to get into. `if ( 'value === $variable ) {`. It's a WP Standard and values are generally shorter than variable & function names. (multiple occurances)

File: aioseop_class.php (hunk 1 above)
Comment (EkoJR) [reply aioseop_class.php]: https://github.com/semperfiwebdesign/all-in-one-seo-pack/blob/53b0a5f9c74941979d645caa4866145287f7e944/admin/aioseop_module_class.php#L302

Comment (arnaudbroes) [reply aioseop_class.php]: Done.

File: aioseop_class.php (hunk 3 above)
Comment (arnaudbroes) [reply aioseop_class.php]: Done.

File: aioseop_class.php (hunk 2 above)
Comment (arnaudbroes) [reply aioseop_class.php]: Done.

**Comment by arnaudbroes** [comment]
@EkoJR I've addressed all comments. Now I am waiting on either you or @contactashish13 to help me finalize my PHPUnit setup so I can add the unit tests that were requested.

**Comment by arnaudbroes** [comment]
@EkoJR this can be code-reviewed again. I implemented all the comments and added the required unit tests. I'm going to expand those unit tests to include all pages (homepage, search page, 404 page, ...) and also test all scenario's (homepage title, aioseop title, homepage as static page, etc ..)

**Pull request merged**